In [2]:
import csv
f = open("guns.csv",'r')
dataset = csv.reader(f)
data = list(dataset)
print(data[:5])

[['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education'], ['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4']]


In [3]:
headers = data[0]
data = data[1:]
print(headers)
print(data[:5])

['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education']
[['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4'], ['5', '2012', '02', 'Suicide', '0', 'M', '31', 'White', '100', 'Other specified', '2']]


In [4]:
years = [each[1] for each in data]
year_counts = {}
for each in years:
    if each in year_counts:
        year_counts[each] += 1
    else:
        year_counts[each] = 1
print(year_counts)

{'2012': 33563, '2014': 33599, '2013': 33636}


In [15]:
import datetime
dates = [datetime.datetime(year = int(each[1]), month = int(each[2]), day = 1) for each in data]
print(dates[:5])    
date_counts = {}
for each in dates:
    if each in date_counts:
        date_counts[each] += 1
    else:
        date_counts[each] = 1
print(date_counts)     

[datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0)]
{datetime.datetime(2012, 1, 1, 0, 0): 2758, datetime.datetime(2013, 8, 1, 0, 0): 2859, datetime.datetime(2014, 1, 1, 0, 0): 2651, datetime.datetime(2012, 2, 1, 0, 0): 2357, datetime.datetime(2012, 8, 1, 0, 0): 2954, datetime.datetime(2012, 4, 1, 0, 0): 2795, datetime.datetime(2014, 4, 1, 0, 0): 2862, datetime.datetime(2013, 4, 1, 0, 0): 2798, datetime.datetime(2014, 11, 1, 0, 0): 2756, datetime.datetime(2012, 3, 1, 0, 0): 2743, datetime.datetime(2012, 12, 1, 0, 0): 2791, datetime.datetime(2014, 3, 1, 0, 0): 2684, datetime.datetime(2014, 10, 1, 0, 0): 2865, datetime.datetime(2012, 9, 1, 0, 0): 2852, datetime.datetime(2014, 7, 1, 0, 0): 2884, datetime.datetime(2014, 2, 1, 0, 0): 2361, datetime.datetime(2013, 2, 1, 0, 0): 2375, datetime.datetime(2012, 5, 1, 0, 0): 2999, datetime.datetime(2014, 5, 1, 0, 0):

In [5]:
sex_column = [each[5] for each in data] 

sex_counts = {}
for row in sex_column:
    if row in sex_counts:
        sex_counts[row] += 1
    else:
        sex_counts[row] = 1
print(sex_counts)     

{'F': 14449, 'M': 86349}


In [6]:
race_column = [each[7] for each in data] 

race_counts = {}
for row in race_column:
    if row in race_counts:
        race_counts[row] += 1
    else:
        race_counts[row] = 1
print(race_counts)     

{'Asian/Pacific Islander': 1326, 'White': 66237, 'Black': 23296, 'Hispanic': 9022, 'Native American/Native Alaskan': 917}


## Gun death 
#Read the dataset in as a list using the csv module.
#Open the file using the open() function.
#Use the csv.reader() function to load the opened file.
#Call list() on the result to get a list of all the data in the file.
#Assign the result to the variable data.
#Display the first 5 rows of data to verify everything.
#Extract the first row of data, and assign it to the variable headers.
#Remove the first row from data.
#Display headers.
#Display the first 5 rows of data to verify that you removed the header row properly.
#Use a list comprehension to extract the year column from data.
#Because the year column is the second column in the data, you'll need to get the element at index 1 in each row.
#Assign the result to the variable years.
#Create an empty dictionary called year_counts.
#Loop through each element in years.
#If the element isn't a key in year_counts, create it, and set the value to 1.
#If the element is a key in year_counts, increment the value by one.
#Display year_counts to see how many gun deaths occur in each year.
#Use a list comprehension to create a datetime.datetime object for each row. Assign the result to dates.
#The year column is in the second element in each row.
#The month column is the third element in each row.
#Make sure to convert year and month to integers using int().
#Pass year, month, and day=1 into the datetime.datetime() function.
#Display the first 5 rows in dates to verify everything worked.
#Count up how many times each unique date occurs in dates. Assign the result to date_counts.
#This follows a similar procedure to what we did in the last screen with year_counts.
#Display date_counts.


In [7]:
f = open("census.csv",'r')
census_beta = csv.reader(f)
census = list(census_beta)
print(census)

[['Id', 'Year', 'Id', 'Sex', 'Id', 'Hispanic Origin', 'Id', 'Id2', 'Geography', 'Total', 'Race Alone - White', 'Race Alone - Hispanic', 'Race Alone - Black or African American', 'Race Alone - American Indian and Alaska Native', 'Race Alone - Asian', 'Race Alone - Native Hawaiian and Other Pacific Islander', 'Two or More Races'], ['cen42010', 'April 1, 2010 Census', 'totsex', 'Both Sexes', 'tothisp', 'Total', '0100000US', '', 'United States', '308745538', '197318956', '44618105', '40250635', '3739506', '15159516', '674625', '6984195']]


In [8]:
mapping = {'Asian/Pacific Islander': 15834141,'Black': 40250635,'Hispanic': 44618105,'Native American/Native Alaskan': 3739506,'White':197318956}                 
race_per_hundredk = {}
for each in race_counts:
    race_per_hundredk[each] = (race_counts[each]/mapping[each])*100000
print(race_per_hundredk)    

{'Asian/Pacific Islander': 8.374309664161762, 'White': 33.56849303419181, 'Black': 57.8773477735196, 'Hispanic': 20.220491210910907, 'Native American/Native Alaskan': 24.521955573811088}


In [10]:
intents = [each[3] for each in data]
races = [each[7] for each in data]
homicide_race_counts = {}
for i,race in enumerate(races):
    if intents[i] == 'Homicide':
       if race in homicide_race_counts:
          homicide_race_counts[race] += 1
       else:
          homicide_race_counts[race] = 1
print(homicide_race_counts)
            

{'White': 9147, 'Asian/Pacific Islander': 559, 'Black': 19510, 'Hispanic': 5634, 'Native American/Native Alaskan': 326}


## Findings
Data of homicide by race counts per  100000
'White': 9147, 'Asian/Pacific Islander': 559, 'Black': 19510, 'Hispanic': 5634, 'Native American/Native Alaskan': 326
Black people die most due to gun homicide.

##That's it for the guided steps! We recommend exploring the data more on your own.

Here are some potential next steps:

Figure out the link, if any, between month and homicide rate.
Explore the homicide rate by gender.
Explore the rates of other intents, like Accidental, by gender and race.
Find out if gun death rates correlate to location and education.
We recommend creating a Github repository and placing this project there. It will help other people, including employers, see your work. As you start to put multiple projects on Github, you'll have the beginnings of a strong portfolio. You're welcome to keep working on the project here, but we recommend downloading it to your computer using the download icon above and working on it there.